### DISCUSSION TITLE
How to specify dynamic library path or other environment parameters before test start?

### DISCUSSION THREAD

**Discussion by liufuyang** [xunit/xunit#2296] [Question] [answered]
I have encountered a problem like this:
https://blog.bokan.io/posts/aesccm-and-aesgcm-in-dotnet-core-on-macos-catalina/
Basically the solution needs me to:
`set the DYLD_LIBRARY_PATH in both Unit Test and Run/Debug configurations` so to be able to dynamically load `libssl `.

But I find no solution to test an environment parameter before xUnit start the test.

Do you have some suggestions?

**Comment by bradwilson** [comment, marked as answer]
What test runner are you using? I assume `dotnet test` (which is VSTest) inherits the environment variables from the console that it was run from, so setting those environments variable(s) in the console before launching `dotnet test` should suffice. This is probably a question for the VSTest team: https://github.com/microsoft/vstest

**Reply by liufuyang** [reply]
Thanks. It is `xUnit` I was using. Test was launched in visual studio on Mac.
